src/nvapi_gpu.cpp

```diff
@@ -486,6 +486,45 @@ extern "C" {
         return Ok(n);
     }
 
+    NvAPI_Status __cdecl NvAPI_GPU_GetGPUInfo(NvPhysicalGpuHandle hPhysicalGpu, NV_GPU_INFO* pGpuInfo) {
+        constexpr auto n = __func__;
+
+        if (nvapiAdapterRegistry == nullptr)
+            return ApiNotInitialized(n);
+
+        if (pGpuInfo == nullptr)
+            return InvalidArgument(n);
+
+        if (pGpuInfo->version != NV_GPU_INFO_VER1 && pGpuInfo->version != NV_GPU_INFO_VER2)
+            return IncompatibleStructVersion(n);
+
+        auto adapter = reinterpret_cast<NvapiAdapter*>(hPhysicalGpu);
+        if (!nvapiAdapterRegistry->IsAdapter(adapter))
+            return ExpectedPhysicalGpuHandle(n);
+
+        auto architectureId = adapter->GetArchitectureId();
+
+        switch (pGpuInfo->version) {
+            case NV_GPU_INFO_VER1: {
+                auto pGpuInfoV1 = reinterpret_cast<NV_GPU_INFO_V1*>(pGpuInfo);
+                *pGpuInfoV1 = {};
+                break;
+            }
+            case NV_GPU_INFO_VER2:
+                *pGpuInfo = {};
```

Comment:
We should probably preserve `pGpuInfo->version` as was passed on input, right? I'd just set `rayTracingCores` and `tensorCores` in this branch and (also in VER1 branch above) leave version and reserved members untouched.

Reply:
`V1` doesn't have those fields (it only has `bIsExternalGpu` and a reserved flag), but good point actually about clearing the version info. We do this at several places though, so that will be a bigger change. 

That said, even Adam used `memset` to clear a struct (https://github.com/adamnv/dxvk-nvapi/blob/c859f94b4213c5148f89a785931448d4630871e2/src/nvapi_d3d.cpp#L63) ;). So I guess in reality it would make no difference. May be something to consider for the future?

Reply:
Oh well then, there's probably nothing that checks the returned version anyway.